apps/sms/views/shared/js/utils.js

```diff
@@ -473,6 +473,25 @@
       });
       return parsed;
     },
+    url(base, params) {
+      if (base.indexOf('?') === -1) {
+        base += '?';
+      } else {
+        base += '&';
+      }
+
+      for (var key in params) {
+        if (params[key] == null) { // null or undefined
+          continue;
+        }
+
+        base +=
+          encodeURIComponent(key) + '=' +
+          encodeURIComponent(params[key]) + '&';
```

Comment:
I think that what we could do for now under this architecture. Using draft id sounds good for me but not for bookmark issue. Just because we can reuse the current draft functionality instead of invent another id. But this means we'll need to access draft in activity service, and we might need to reconsider again whether we'll need to provide more method to query draft by draft service.

BTW about the page bookmark, I don't know if it reasonable to allow user to ping the draft, especially for the new message view.  But it's another story anyway.